packages/react-aria-components/src/utils.tsx

```diff
@@ -136,19 +136,25 @@ export interface SlotProps {
   slot?: string
 }
 
-export function useContextProps<T, U extends SlotProps, E extends Element>(props: T & SlotProps, ref: React.ForwardedRef<E>, context: React.Context<ContextValue<U, E>>): [T, React.RefObject<E>] {
-  let ctx = useContext(context) || {};
-  if ('slots' in ctx && ctx.slots) {
-    if (!props.slot && !ctx.slots[defaultSlot]) {
+export function useSlottedContext<U extends SlotProps, E extends Element>(context: React.Context<ContextValue<U, E>>, slot?: string): WithRef<U, E> | null | undefined {
```

Comment:
Split useContextProps into two hooks so that you can consume from one of our contexts that supports slots but without merging with some other props/ref. Used in Calendar/RangeCalendar examples.